src/eth_provider/starknet/kakarot_core.rs

```diff
@@ -132,13 +132,26 @@ pub fn to_starknet_transaction(
     let mut signed_data = Vec::with_capacity(transaction.transaction.length());
     transaction.transaction.encode_without_signature(&mut signed_data);
 
+    // Signed data length after RLP encoding (used for calldata length check)
+    let signed_data_len = signed_data.len();
+
+    // Pack the calldata in 31-byte chunks
+    let mut signed_data: Vec<FieldElement> = std::iter::once(FieldElement::from(signed_data_len))
+        .chain(
+            signed_data
+                .as_slice()
+                .chunks(31)
+                .filter_map(|chunk_bytes| FieldElement::from_byte_slice_be(chunk_bytes).ok()),
+        )
+        .collect();
+
     // Prepare the calldata for the Starknet invoke transaction
     let capacity = 6 + signed_data.len();
 
     // Check if call data is too large
     #[cfg(not(feature = "hive"))]
-    if capacity > *MAX_FELTS_IN_CALLDATA {
```

Comment:
can't you use capacity? it seems to be the same

Reply:
Not really because the capacity takes into account the `signed_data` after packing and therefore the size is not the same. Whereas here I put the `signed_data_len` before the packing in order to have the real length just after the RLP encoding and before packing.

Reply:
ahhhh ok ok but I think there is an issue here: we want to compare `MAX_FELTS_IN_CALLDATA` with the actual calldata we send to Starknet right? So that would be capacity, whereas signed_data_len is prepacked and doesn't represent what we actually send to Starknet

Reply:
Just fixed as per our discussion, for further clarification on the Felt limitation: https://docs.starknet.io/tools/limits-and-triggers/